imap_processing/cli.py

```diff
@@ -147,63 +255,85 @@ class Codice(ProcessInstrument):
 
     def process(self):
         """Perform CoDICE specific processing."""
-        print(f"Processing CoDICE {self.level}")
+        print(f"Processing CoDICE {self.data_level}")
 
 
 class Glows(ProcessInstrument):
     """Process GLOWS."""
 
     def process(self):
         """Perform GLOWS specific processing."""
-        print(f"Processing GLOWS {self.level}")
+        print(f"Processing GLOWS {self.data_level}")
 
 
 class Hi(ProcessInstrument):
     """Process IMAP-Hi."""
 
     def process(self):
         """Perform IMAP-Hi specific processing."""
-        print(f"Processing IMAP-Hi {self.level}")
+        print(f"Processing IMAP-Hi {self.data_level}")
 
 
 class Hit(ProcessInstrument):
     """Process HIT."""
 
     def process(self):
         """Perform HIT specific processing."""
-        print(f"Processing HIT {self.level}")
+        print(f"Processing HIT {self.data_level}")
 
 
 class Idex(ProcessInstrument):
     """Process IDEX."""
 
     def process(self):
         """Perform IDEX specific processing."""
-        print(f"Processing IDEX {self.level}")
+        print(f"Processing IDEX {self.data_level}")
 
 
 class Lo(ProcessInstrument):
     """Process IMAP-Lo."""
 
     def process(self):
         """Perform IMAP-Lo specific processing."""
-        print(f"Processing IMAP-Lo {self.level}")
+        print(f"Processing IMAP-Lo {self.data_level}")
 
 
 class Mag(ProcessInstrument):
     """Process MAG."""
 
     def process(self):
         """Perform MAG specific processing."""
-        print(f"Processing MAG {self.level}")
+        print(f"Processing MAG {self.data_level}")
+        file_paths = self.download_dependencies()
+
+        if self.data_level == "l1a":
+            # File path is expected output file path
+            if len(file_paths) > 1:
+                raise ValueError(
+                    f"Unexpected dependencies found for MAG L1A:"
+                    f"{file_paths}. Expected only one dependency."
+                )
+            filename = imap_data_access.ScienceFilePath.generate_from_inputs(
+                "mag", "l1a", "raw", self.start_date, self.end_date, self.version
+            )
+            mag_l1a(file_paths[0], filename.construct_path())
+            print(f"Generated file: {filename.construct_path()}")
+
+            if self.upload_to_sdc:
+                print(f"Uploading file from: {filename.construct_path()}")
+                # TODO: figure out data_dir, because now this fails.
+                #  Should switch to using IMAP_DATA_DIR env var.
+                imap_data_access.upload(filename.construct_path())
 
 
 class Swapi(ProcessInstrument):
     """Process SWAPI."""
 
     def process(self):
         """Perform SWAPI specific processing."""
-        print(f"Processing SWAPI {self.level}")
+        print(f"Processing SWAPI {self.data_level}")
+        # for dep in self.dependency:
+        #     imap_data_access.query()
```

Comment:
self.download_dependencies()? Or just remove the comments for now if not needed.